server/routes/paymentCalculation.js

```diff
@@ -0,0 +1,42 @@
+const actionService = require('../services/actionsService')
+const parcelService = require('../services/parcelService')
+const paymentCalculationService = require('../services/paymentCalculationService')
+const schema = require('../schema/paymentCalculation')
+
+module.exports = {
+  method: 'POST',
+  path: '/payment-calculation',
+  options: {
+    validate: {
+      payload: schema,
+      failAction: async (request, h, error) => {
+        console.log('rejected payload', request.payload)
+        return h.response().code(400).takeover()
+      }
+    },
+    handler: async (request, h) => {
+      const requestedActions = request.payload.actions
```

Comment:
(A very small point)

This could be replaced by ` const { actions: requestedActions, parcelRef } = request.payload`.

Reply:
Good spot, thanks.

Reply:
I have made that change.